src/app/application-frame/application-frame.component.ts

```diff
@@ -16,14 +17,21 @@ import { UserTelemetryOrchestrationService } from '../shared/telemetry/user-tele
     </ht-application-header>
     <div class="app-body">
       <ht-navigation class="left-nav"></ht-navigation>
-      <div class="app-content">
+      <div class="app-content" *ngIf="this.timeRangeHasInit$ | async">
         <router-outlet></router-outlet>
       </div>
     </div>
   `
 })
 export class ApplicationFrameComponent implements OnInit {
-  public constructor(private readonly userTelemetryOrchestrationService: UserTelemetryOrchestrationService) {}
+  public readonly timeRangeHasInit$: Observable<TimeRange>;
+
+  public constructor(
+    private readonly userTelemetryOrchestrationService: UserTelemetryOrchestrationService,
+    private readonly timeRangeService: TimeRangeService
+  ) {
+    this.timeRangeHasInit$ = this.timeRangeService.getTimeRangeAndChanges();
```

Comment:
nit: could take(1) here, will let this clean up rather than hanging around

Reply:
Wouldn't that defeat the purpose ? it will take the first emission which will be undefined, and never show the app content

Reply:
`getTimeRangeAndChanges` should never output an undefined - it just won't emit until a time range is assigned. If we broke that, that means it's mistyped and we should go fix.